src/OrchardCore.Modules/OrchardCore.Facebook/Drivers/FacebookCoreSettingsDisplayDriver.cs

```diff
@@ -58,39 +54,28 @@ public override async Task<IDisplayResult> EditAsync(FacebookCoreSettings settin
 
             return Initialize<FacebookCoreSettingsViewModel>("FacebookCoreSettings_Edit", model =>
             {
+                var protector = _dataProtectionProvider.CreateProtector(FacebookConstants.Features.Core);
+
                 model.AppId = settings.AppId;
-                model.AppSecret = settings.AppSecret;
+                model.AppSecret = protector.Unprotect(settings.AppSecret);
```

Comment:
`try/catch` like you did for SMTP and OpenID settings, to avoid decryption exceptions?

Reply:
It's not necessary in drivers, as they are themselves skipped on exceptions and the exception is logged. It is necessary during the configuration phase though, so I will assume that's what you meant.